client/components/ChatInput.vue

```diff
@@ -260,34 +260,73 @@ export default {
 				autocompletionRef.hide();
 			}
 
-			this.channel.inputHistoryPosition = 0;
-			this.channel.pendingMessage = "";
-			this.$refs.input.value = "";
-			this.setInputSize();
+			const resetInput = () => {
+				this.channel.inputHistoryPosition = 0;
+				this.channel.pendingMessage = "";
+				this.$refs.input.value = "";
+				this.setInputSize();
+			};
+
+			const sendMessage = () => {
+				resetInput();
+
+				// Store new message in history if last message isn't already equal
+				if (this.channel.inputHistory[1] !== text) {
+					this.channel.inputHistory.splice(1, 0, text);
+				}
 
-			// Store new message in history if last message isn't already equal
-			if (this.channel.inputHistory[1] !== text) {
-				this.channel.inputHistory.splice(1, 0, text);
-			}
+				// Limit input history to 100 entries
+				if (this.channel.inputHistory.length > 100) {
+					this.channel.inputHistory.pop();
+				}
 
-			// Limit input history to a 100 entries
-			if (this.channel.inputHistory.length > 100) {
-				this.channel.inputHistory.pop();
-			}
+				if (text[0] === "/") {
+					const args = text.substr(1).split(" ");
+					const cmd = args.shift().toLowerCase();
+
+					if (
+						Object.prototype.hasOwnProperty.call(commands, cmd) &&
+						commands[cmd].input(args)
+					) {
+						return false;
+					}
+				}
 
-			if (text[0] === "/") {
-				const args = text.substr(1).split(" ");
-				const cmd = args.shift().toLowerCase();
+				socket.emit("input", {target, text});
+			};
+
+			if (this.$store.state.serverConfiguration.fileUpload) {
+				const lines = 1 + (text.match(/\n/g) || "").length;
+
+				if (lines > 3 || text.length > 700) {
```

Comment:
what happens when someone pastes more than 700 lines?

Reply:
the same thing that happens for text with more than 3?

lines > 3 || text.length > 700

means "either over 3 lines or text longer than 700 characters"

Not sure I understand your issue

Reply:
well, reading sure is hard.
sorry for the noise